src/Tank.cpp

```diff
@@ -55,17 +55,43 @@ void Tank::set_current_speed(float speed) { mCurrentSpeed = speed; }
 sf::Vector2f Tank::get_position() { return mPos; }
 
 void Tank::update() {
-  const auto rotation_degree = mBody.get_rotation() - 90;
-  const auto rotation_radians = pi / 180.f * rotation_degree;
-
-  mPos.x += mCurrentSpeed * std::cos(rotation_radians);
-  mPos.y += mCurrentSpeed * std::sin(rotation_radians);
+  if(!mBorder)
+  {
+    const auto rotation_degree = mBody.get_rotation() - 90;
+    const auto rotation_radians = pi / 180.f * rotation_degree;
+
+    mPos.x += mCurrentSpeed * std::cos(rotation_radians);
+    mPos.y += mCurrentSpeed * std::sin(rotation_radians);
+  };
 }
 
 float Tank::get_tower_rotation() const { return mTower.get_rotation(); }
 
 void Tank::draw(sf::RenderWindow &window) {
+  check_border();
   update();
   mBody.draw(window, mPos.x, mPos.y);
   mTower.draw(window, mPos.x, mPos.y);
 }
+
+void Tank::check_border()
```

Comment:
This method should return boolean, it can be renamed to somethig that starts with "is_"